### PR TITLE
Refactoring- Removed 3 design smells

### PR DESCRIPTION
**Pull request by ashishbhasin45**
1. Refactored to resolve cyclic dependency
Location- 	
Sub-Project: imixs-workflow-engine
     	Package: org.imixs.workflow.engine
     	Classes: AsyncEventScheduler and AsyncEventService

2. Refactoring applied- Move method to increase cohesion.
Location(s)-   Sub-project: imixs-workflow-jax-rs
     Package: org.imixs.workflow.jaxrs
     Class: ModelRestService

     Sub-project: imixs-workflow-engine
     Package: org.imixs.workflow.engine
     Class: ModelService

     Sub-project: imixs-workflow-core
     Package: org.imixs.workflow
     Class: ModelManager
ModelRestService was using ModelService’s methods quite a few times in its method printVersionTable. I moved the method to perform the desired operation in the ModelService where it belonged and then called that method once in the ModelRestService to get the output.
The model was also defined in the ModelManager interface as ModelService implements it.

3. Refactoring applied- Replace conditional with polymorphism.
Location(s)-   Sub-project: imixs-workflow-faces
     Package: org.imixs.workflow.faces.util
     Class: ErrorHandler

     Sub-project: imixs-workflow-faces
     Package: org.imixs.workflow.faces.util
     Class: ValidationException

    Sub-project: imixs-workflow-core
    Package: org.imixs.workflow.exceptions
    Class: PluginException

    Sub-project: imixs-workflow-core
    Package: org.imixs.workflow.exceptions
    Class: WorkflowException
ErrorHandler class had conditionals in the addErrorMessage where it was checking the instance of PluginException and ValidationException and performing action based on that. I created a method formatErrorMessageWithParameters in the abstract class workflowException and overrode this method in PluginException and ValidationException child classes. In the ErrorHandler class I changed the conditional with the call to this new method and based on the object type it will perform desired behavior.

### CODE REVIEW

File: imixs-workflow-engine/src/main/java/org/imixs/workflow/engine/AsyncEventSchedulerConfig.java
Comment (rsoika) [inline imixs-workflow-engine/src/main/java/org/imixs/workflow/engine/AsyncEventSchedulerConfig.java:1]: ok

File: imixs-workflow-engine/src/main/java/org/imixs/workflow/engine/AsyncEventScheduler.java
Comment (rsoika) [inline imixs-workflow-engine/src/main/java/org/imixs/workflow/engine/AsyncEventScheduler.java:1]: ok

File: imixs-workflow-engine/src/main/java/org/imixs/workflow/engine/AsyncEventService.java
Comment (rsoika) [inline imixs-workflow-engine/src/main/java/org/imixs/workflow/engine/AsyncEventService.java:1]: ok

File: imixs-workflow-faces/src/main/java/org/imixs/workflow/faces/util/ErrorHandler.java
Comment (rsoika) [inline imixs-workflow-faces/src/main/java/org/imixs/workflow/faces/util/ErrorHandler.java:1]: ok

File: imixs-workflow-faces/src/main/java/org/imixs/workflow/faces/util/ValidationException.java
Comment (rsoika) [inline imixs-workflow-faces/src/main/java/org/imixs/workflow/faces/util/ValidationException.java:1]: lets move this method into the abstract class `WorkflowException`

File: imixs-workflow-core/src/main/java/org/imixs/workflow/exceptions/PluginException.java
Comment (rsoika) [inline imixs-workflow-core/src/main/java/org/imixs/workflow/exceptions/PluginException.java:1]: lets move this method into the abstract class `WorkflowException`

File: imixs-workflow-core/src/main/java/org/imixs/workflow/BPMNRuleEngine.java
Comment (rsoika) [inline imixs-workflow-core/src/main/java/org/imixs/workflow/BPMNRuleEngine.java:1]: remove this as we can have a valid implementation directly in the abstract `WorkflowException` class

File: imixs-workflow-core/src/main/java/org/imixs/workflow/exceptions/WorkflowException.java
Comment (rsoika) [inline imixs-workflow-core/src/main/java/org/imixs/workflow/exceptions/WorkflowException.java:1]: just put the implementation here

File: imixs-workflow-jax-rs/src/main/java/org/imixs/workflow/jaxrs/ModelRestService.java
Comment (rsoika) [inline imixs-workflow-jax-rs/src/main/java/org/imixs/workflow/jaxrs/ModelRestService.java:1]: I understand your attention, but lets have the method `modelVersionTableToString` still in the `ModelRestService` as this is a very specific part of the REST API

**Review by rsoika — CHANGES_REQUESTED** [review]
That's amazing! Great and thanks again for your contribution :1st_place_medal: 

I have just two comments:

 - We can place the method `formatErrorMessageWithParameters` directly into the `WorkflowException`. This is a good place and your method makes of course sense. And so we can leave the other Exceptions Types as the are
 
 - Place the method `modelVersionTableToString` into the `ModelRestService` as this is a very specific part of the REST API. The `ModelService` in the Engine package should not handle this part.

File: imixs-workflow-jax-rs/src/main/java/org/imixs/workflow/jaxrs/ModelRestService.java
Comment (ashishbhasin45) [reply imixs-workflow-jax-rs/src/main/java/org/imixs/workflow/jaxrs/ModelRestService.java:1]: Moved the function to modelRestService.

File: imixs-workflow-core/src/main/java/org/imixs/workflow/BPMNRuleEngine.java
Comment (ashishbhasin45) [reply imixs-workflow-core/src/main/java/org/imixs/workflow/BPMNRuleEngine.java:1]: This was the method which was defined in interface modelManager and overriden in  model service, now I have moved it to modelRestservice as per your suggestion, and it no longer a part of the modelManager hence not required to be overriden here.

**Review by rsoika — APPROVED** [review]
ok

**Review by rsoika — APPROVED** [review]

**Review by rsoika — APPROVED** [review]

**Review by rsoika — APPROVED** [review]

File: imixs-workflow-core/src/main/java/org/imixs/workflow/exceptions/PluginException.java
Comment (rsoika) [reply imixs-workflow-core/src/main/java/org/imixs/workflow/exceptions/PluginException.java:1]: I changed this now in the Issue #857 . This simplifies the code much more and we have the feature now for all Imixs Exception Types. 
Thanks for you contribution!

**Pull request merged**